keras_nlp/models/gpt2/gpt2_backbone.py

```diff
@@ -170,6 +174,116 @@ def __init__(
         self.intermediate_dim = intermediate_dim
         self.dropout = dropout
         self.max_sequence_length = max_sequence_length
+        self.transformer_layers = []
+        for i in range(self.num_layers):
+            self.transformer_layers.append(
+                self.get_layer(f"transformer_layer_{i}")
+            )
+        self.layer_norm = self.get_layer("layer_norm")
+        self.token_embeddings = self.get_layer("token_embedding")
+        self.position_embeddings = self.get_layer("position_embedding")
+        self.embeddings_dropout = self.get_layer("embeddings_dropout")
+
+    def call_with_cache(self, inputs, cache, cache_index=None):
```

Comment:
I think it might be simpler to just blow up all the inputs here as explicit args, e.g `token_ids, padding_mask, cache, cache_index`

Mixing a dict for some inputs and not for others is a bit confusing.

Reply:
nice idea.